cards.js

```diff
@@ -0,0 +1,325 @@
+                script.setAttribute('data-card', cardId);
+                script.src = `${cardPath}/${jsFile}`;
+                document.head.appendChild(script);
+                
+                loadedScripts.add(cardId);
+                console.log(`Loaded script for card: ${cardId}`);
+            }
+        } catch (error) {
+            console.warn(`Failed to load script for card ${cardId}:`, error);
+        }
+    }
+
+    /**
+     * Render cards for a specific role
+     * @param {string} role - The role to render cards for
+     * @param {string} containerId - ID of the container to render cards into
+     */
+    async function renderCardsForRole(role, containerId = 'cards-container') {
+        const container = document.getElementById(containerId);
+        if (!container) {
+            console.error(`Cards container not found: ${containerId}`);
+            return;
+        }
+
+        // Clear existing cards
+        container.innerHTML = '';
+
+        // Get card definitions for this role
+        const cardDefs = getCardsForRole(role);
+        if (!cardDefs || cardDefs.length === 0) {
+            // Hide container if no cards
+            container.style.display = 'none';
+            return;
+        }
+
+        // Show container if we have cards
+        container.style.display = 'block';
+
+        try {
+            // Load and render each card
+            for (const cardDef of cardDefs) {
+                const cardData = await loadCard(cardDef.id);
+                
+                const cardWrapper = document.createElement('div');
+                cardWrapper.className = 'card-wrapper';
+                cardWrapper.setAttribute('data-card-id', cardDef.id);
+                cardWrapper.innerHTML = cardData.html;
+                container.appendChild(cardWrapper);
+            }
+
+            // Cards are now rendered and ready for persistence
+            console.log(`Rendered ${cardDefs.length} cards for role: ${role}`);
+            
+            // Note: Persistence refresh is handled by the caller
+
+        } catch (error) {
+            console.error('Error rendering cards:', error);
+            container.innerHTML = '<div class="card card-error"><p>Error loading cards</p></div>';
+        }
+    }
+
+    /**
+     * Get list of card definitions for a role from cards data
+     * @param {string} role - The role to get cards for
+     * @returns {Array} Array of card objects with id and path
+     */
+    function getCardsForRole(role) {
+        if (!window.cardsData || !window.cardsData[role]) {
+            return [];
+        }
+
+        return window.cardsData[role] || [];
+    }
+
+    /**
+     * Add a card to a role (useful for moves that grant cards)
+     * @param {string} role - The role to add the card to
+     * @param {string|object} cardIdOrDef - Card ID (string) or card definition object {id, path}
+     */
+    function addCardToRole(role, cardIdOrDef) {
```

Comment:
Hmn suspicious... shouldn't need to add card to role - instead it just renders below move